src/redux/fallbackExplorer.js

```diff
@@ -357,8 +365,34 @@ export const fallbackExplorerInit = () => async (dispatch, getState) => {
       },
       type: FALLBACK_EXPLORER_SET_HANDLERS,
     });
-  };
-  fetchAssetsBalancesAndPrices();
+  }
+};
+
+export const fallbackExplorerInit = () => async (dispatch, getState) => {
+  const { accountAddress, network } = getState().settings;
+  const { latestTxBlockNumber, mainnetAssets } = getState().fallbackExplorer;
+  const { coingeckoIds } = getState().additionalAssetsData;
+  // If mainnet, we need to get all the info
+  // 1 - Coingecko ids
+  // 2 - All tokens list
+  // 3 - Etherscan token transfer transactions
+  if (NetworkTypes.mainnet === network) {
+    const newMainnetAssets = await findAssetsToWatch(
+      accountAddress,
+      latestTxBlockNumber,
+      dispatch,
+      coingeckoIds
+    );
+
+    await dispatch({
+      payload: {
+        mainnetAssets: mainnetAssets.concat(newMainnetAssets),
```

Comment:
this existed already but is this right to just do a straight concat of the mainnetAssets with newMainnetAssets without any deduping / uniqueBy?